src/NUnitEngine/nunit.engine/RuntimeFramework.cs

```diff
@@ -608,24 +627,25 @@ private static void AppendOlderDotNetFrameworkVersion(List<RuntimeFramework> fra
         // microsoft will do in the future
         private static void AppendDotNetFourFrameworkVersions(List<RuntimeFramework> frameworks, RegistryKey versionKey)
         {
-            foreach (string profile in new string[] { "Full", "Client" })
+            foreach (Profile profile in new Profile[] { Profile.Full, Profile.Client })
             {
-                var profileKey = versionKey.OpenSubKey(profile);
+                var profileKey = versionKey.OpenSubKey(profile.ToString());
                 if (profileKey == null) continue;
                 
                 if (CheckInstallDword(profileKey))
                 {
                     var framework = new RuntimeFramework(RuntimeType.Net, new Version(4, 0));
-                    framework.DisplayName += " - " + profile;
+                    framework.DisplayName += " - " + profile.ToString();
+                    framework.Profile = profile;
                     frameworks.Add(framework);
 
                     var release = (int)profileKey.GetValue("Release", 0);
                     if (release > 0)
                     {
                         framework = new RuntimeFramework(RuntimeType.Net, new Version(4, 5));
```

Comment:
OK, I missed that - you're right, it's not clear, but rewriting would add a lot of duplication. Maybe just add a comment in front of the return about why we can return at that point.